app/serializers/queued_post_serializer.rb

```diff
@@ -61,10 +61,8 @@ def include_can_delete_user?
   def who_did(status)
     object.
       reviewable_histories.
-      where(
-        reviewable_history_type: ReviewableHistory.types[:transitioned],
-        status: Reviewable.statuses[status]
-      ).
+      transitioned.
+      public_send(status).
```

Comment:
Is there a way to avoid `public_send`? That's asking for trouble 😛

Reply:
Well it allows to use the automatically defined scopes from AR and if the status isn’t a valid one for whatever reason it will break. `#who_did` is a protected method and is only used in two places (L36 and L40), we control the value of `status`, nothing inherits from `QueuedPostSerializer` so I guess the trouble should be quite contained :slightly_smiling_face: Wdyt?

Reply:
If it's used in just two places, with `:approved` and `:rejected` as args, I would consider inlining this method. But I guess I'm fine with it as is too. 🙂 

Oh, or a half-way inlining solution would work too, e.g.

```rb
def approved_by_id
  scope.approved.last&.created_by_id
end

def rejected_by_id
  scope.rejected.last&.created_by_id
end

protected

# not necessarily the best name 😉
def scope
  object.
    reviewable_histories.
    transitioned.
    order(:created_at)
end
```

Reply:
Updated! I wasn’t that inspired either and went with `#post_history` for the name.